control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -3989,8 +3997,15 @@ func (r *HostedControlPlaneReconciler) validateAWSKMSConfig(ctx context.Context,
 
 	token, err := util.CreateTokenForServiceAccount(ctx, manifests.KASContainerAWSKMSProviderServiceAccount(), guestClient)
 	if err != nil {
-		// service account might not be created in the guest cluster yet.
-		log.Error(err, "failed to create token for KMS provider service account")
```

Comment:
Should we delete the log trace? I mean, it's great to have the condition in the status but I think to keep the log entry will help on debugging.

Reply:
I don't think we need it, If we get an error here:
1. either the service account is not created yet, so the condition will simply reconcile very fast to true once it is created.
2. or the KAS is not working/reachable which means the `Available` condition of the HCP and HC will be false, and any debugging would need to be done on the KAS deployment/logs. Logging that the KAS is not reachable here is kind redundant.